src/main.c

```diff
@@ -1790,6 +1796,17 @@ static int parse_configured_connectors(yubihsm_context *ctx, char **connectors,
 }
 
 #ifdef USE_ASYMMETRIC_AUTH
+
+static void free_configured_pubkeys(yubihsm_context *ctx) {
+  if (ctx->device_pubkey_list) {
```

Comment:
similarly `if (ctx && ctx->device_pubkey_list)`

Reply:
Personally I would then also do it in other similar functions. And I'm also reluctant to maybe inadvertently hide bugs.

Reply:
In this case, with the usage we have, we know it's not needed as well

Reply:
given that it's an internal function it's OK not to check if we can guarantee pre-conditions. The main risk is if something changes in the future it becomes hard to keep track of this kind of things.

Reply:
Yes. But just keeping with the style of similar code around it.